wolfcrypt/src/asn.c

```diff
@@ -10187,7 +10187,7 @@ int wc_OBJ_sn2nid(const char *sn)
         sn = "SECP256R1";
     /* OpenSSL allows lowercase curve names */
     for (i = 0; i < (int)(sizeof(curveName) - 1) && *sn; i++) {
-        curveName[i] = (char)XTOUPPER(*sn++);
+        curveName[i] = (char)XTOUPPER((unsigned int) *sn++);
```

Comment:
The `toupper` should take an `int`. Also oddly there are other places in our code where we cast to `unsigned char`. Curious what the actual warning was here?

Reply:
Warning was:

```
wolfcrypt/src/[asn.c | evp.c] <line#:col#> error: array subscript has type 'char'
Error code 1
```

Reply:
I think it did not like that it was "signed" instead of "unsigned"